src/sage/modular/modform/element.py

```diff
@@ -372,7 +377,7 @@ def coefficients(self, X):
             self.__coefficients
         except AttributeError:
             self.__coefficients = {}
-        if isinstance(X, Integer):
+        if isinstance(X, (int, Integer)):
             X = list(range(1, X + 1))
```

Comment:
```suggestion
        X = Integer(X)
        X = list(range(1, X + 1))
```
This is because the `isinstance` test might fail if the user input something silly like `4/2` which is an integer but SageMath sees it as a rational number. Also, the `X = Integer(X)` will handle the conversion and check at the same time if the input is really an integer, for instance : 
```
sage: Integer('x')
Traceback (most recent call last):
...
TypeError: unable to convert 'x' to an integer
```